synapse/handlers/user_directory.py

```diff
@@ -145,10 +145,6 @@ async def _unsafe_process(self) -> None:
         if self.pos is None:
             self.pos = await self.store.get_user_directory_stream_pos()
 
-        # If still None then the initial background update hasn't happened yet
-        if self.pos is None:
-            return None
```

Comment:
`get_user_directory_stream_pos` always returns an `int` by my reading so this code was unreachable.